lib/Pipeline.js

```diff
@@ -139,6 +139,24 @@ class Pipeline {
     })
   }
 
+  async parseArguments (args) {
+    // is it a list?
+    if (args.out(ns.rdf.first).terms.length === 1) {
```

Comment:
I don't get this, how does it work?

Reply:
`args` is the arguments node. If ref list is used it will have an `rdf:first` property.

```
<step> code:arguments _:args .
_:args rdf:first [] .
_args rdf:rest rdf:nil
```

This is clownface way of checking if the property has any objects

Reply:
Is it possible to check the type `ns.rdf` instead of checking if it has `rdf:first`?

Reply:
Not sure if I understand it correctly, but the list doesn't have an rdf:type, which we could check. Checking for rdf:first is actually the best and only option I can think of. Also we may add a method (`isList`?) to Clownface. But I have to think a little bit more about all the edge cases before I do that, like multiple lists for the same property.

Reply:
I'm not sure I understand how this works either. I'm a bit worried about feeding this with something that, for some reason, has `rdf:first` without being a list. I don't know where `args.` and `ns.` come from though, they could be generated directly in this lib and aren't ever user-provided things which would make this ducktest good enough.

Reply:
I think we are save when we rely on `rdf:first`. If it's not a list, the `rdf:first` triple is used as reference for the first element and no second element is added. Only `args` comes from the user, `ns` is namespace factory defined in the pipeline code.

Also the comment in the [RDF vocab](http://www.w3.org/1999/02/22-rdf-syntax-ns#) for `rdf:first` is very clear:

> The first item in the subject RDF list.

And there is even a constrain in the vocab, so the property can be only used on lists:

`rdf:first rdfs:domain rdf:List .`

So if RDF tools are used to check/generate the triples, users should be guided in the right direction.

Reply:
Good, just wanted to make sure it's safe.